src/main/groovy/life/qbic/oncostore/util/VariantExporter.groovy

```diff
@@ -10,24 +11,37 @@ class VariantExporter {
      * headers for different Variant Call Format versions
      */
     static {
-        vcfHeaders['4.1'] = "##fileformat=VCFv4.1 \n##fileDate=%s\n##source=%s\n##reference=\nCHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\n"
-        vcfHeaders["4.2"] = "##fileformat=VCFv4.2 \n##fileDate=%s\n##source=%s\n##reference=\n#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\n"
+        vcfHeaders['4.1'] = "##fileformat=VCFv4.1 \n##fileDate=%s\n##source=%s\n##reference=%s\nCHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\n"
+        vcfHeaders["4.2"] = "##fileformat=VCFv4.2 \n##fileDate=%s\n##source=%s\n##reference=%s\n#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\n"
     }
 
     /**
      * Generates VCF content from list of variants
      * @param variants the variants to export in VCF
      * @return a VCF content
      */
-    static String exportVariantsToVCF(List<Variant> variants) {
+    static String exportVariantsToVCF(List<Variant> variants, Boolean withConsequences, MetadataContext metadata) {
         def vcfContent = new StringBuilder()
         def date = new Date().format('yyyyMMdd')
-        def vcfHeader = String.format(vcfHeaders["4.1"], date, 'Variantstore')
+
+        // allow to choose VCF version
+        def vcfHeader = String.format(vcfHeaders["4.1"], date, 'variantstore', metadata.referenceGenome.toString())
         vcfContent.append(vcfHeader)
 
+        //determine if SnpEff or VEP
         variants.each { var ->
             vcfContent.append(var.toVcfFormat())
-            vcfContent.append(var.consequences.collect{AnnotationHandler.toSnpEff(it)}.join(","))
+            if (withConsequences) {
+                vcfContent.append(";")
+                if (metadata.variantAnnotation.name == "snpeff") {
+                    vcfContent.append("${AnnotationHandler.AnnotationTools.SNPEFF.tag}=")
+                    vcfContent.append(var.consequences.collect { AnnotationHandler.toSnpEff(it) }.join(","))
+                }
+                else {
```

Comment:
Is there the possibilty of errors in the database? Right now this code would also add an VEP.tag to an entry with a typo such as "NPEFF". If that is a possibility, I'd suggest explicit checks for both variants and an exception else case to handle possible errors.

Reply:
@Steffengreiner thanks for the feedback! good point! I would say that this should/can be check during import while parsing the metadata. Whats your opinion on that?

Reply:
@christopher-mohr Yes i think that would be the best solution! Good Idea 👍

Reply:
@Steffengreiner I added a check for the variant annotation software name (and removed the metadata property VCF files since this is now provided via `files`).